code/go/main-workshop/cdk-workshop.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"github.com/aws/aws-cdk-go/awscdk/v2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awslambda"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsapigateway"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+	"cdk-workshop/hitcounter"
+	"github.com/cdklabs/cdk-dynamo-table-viewer-go/dynamotableviewer"
+)
+
+type CdkWorkshopStackProps struct {
+	awscdk.StackProps
+}
+
+func NewCdkWorkshopStack(scope constructs.Construct, id string, props *CdkWorkshopStackProps) awscdk.Stack {
+	var sprops awscdk.StackProps
+	if props != nil {
+		sprops = props.StackProps
+	}
+	stack := awscdk.NewStack(scope, &id, &sprops)
+
+	helloHandler := awslambda.NewFunction(stack, jsii.String("HelloHandler"), &awslambda.FunctionProps{
+		Code: awslambda.Code_FromAsset(jsii.String("lambda"), nil),
+		Runtime: awslambda.Runtime_NODEJS_16_X(),
+		Handler: jsii.String("hello.handler"),
+	})
+
+	hitcounter := hitcounter.NewHitCounter(stack, "HelloHitCounter", &hitcounter.HitCounterProps{
+		Downstream: helloHandler,
+	})
+
+	awsapigateway.NewLambdaRestApi(stack, jsii.String("Endpoint"), &awsapigateway.LambdaRestApiProps{
+		Handler: hitcounter.Handler(),
+	})
+
+	dynamotableviewer.NewTableViewer(stack, jsii.String("ViewHitCounter"), &dynamotableviewer.TableViewerProps{
+		Title: jsii.String("Hello Hits"),
+		Table: hitcounter.Table(),
+	})
+
+	return stack
+}
+
+func main() {
```

Comment:
Make sure the jsii runtime is correctly closed (this needs to be explicit in Go)

```suggestion
func main() {
	defer jsii.Close()
```